Java/jni/native_documentiterator.cc

```diff
@@ -25,6 +25,33 @@ JNIEXPORT jlong JNICALL Java_com_couchbase_cbforest_DocumentIterator_initEnumera
     }
     return (jlong)e;
 }
+
+JNIEXPORT jlong JNICALL Java_com_couchbase_cbforest_DocumentIterator_initEnumerateSomeDocs
+        (JNIEnv *env, jobject self, jlong dbHandle, jobjectArray jdocIDs){
+
+    // Convert jdocIDs, a Java String[], to a C array of C4Slice:
+    jsize n = env->GetArrayLength(jdocIDs);
+    C4Slice docIDs[n];
```

Comment:
This isn't safe — if n is too large, it can overflow the stack. Use malloc instead.